kf/bpf.go

```diff
@@ -81,9 +81,6 @@ func NewBpfProgram(ctx context.Context, program models.BPFProgram, conf *config.
 			progMapFilePath = filepath.Join(conf.BpfMapDefaultPath, ifaceName, program.MapName)
 		} else if program.ProgType == models.TCType {
 			progMapFilePath = filepath.Join(conf.BpfMapDefaultPath, models.TCMapPinPath, ifaceName, program.MapName)
-		} else {
-			log.Error().Msgf("unsupported program type - %s", program.ProgType)
-			return nil
 		}
```

Comment:
Is it suppose to add another check for the ProgType for tracepoints and keep the else for the type not supported?

Reply:
I agree, we should have an explicit list of types we support

Reply:
This block is for defining the `progMapFilePath` for pinning, when we were not supporting probe types. We wanted to explicitly throw errors for other than prog types of tc or xdp. Since we are supporting probes, we don't need prog maps and also we are not chaining. We don't want to error out here in case of probes.